experiments/preprocessing/merge_and_handle_NaNs.py

```diff
@@ -0,0 +1,81 @@
+import pandas as pd
+
+def load_dataset(df_name):
+    if not df_name.endswith('.csv'):
+        raise ValueError("Only CSV files are supported.")
+    try:
+        df = pd.read_csv(df_name)
+        if df.empty or len(df.columns) == 1:
+            raise ValueError(f"Error in file '{df_name}' - empty or incorrect separator.")
+        df.columns = df.columns.str.lower()
+        return df
+    except (FileNotFoundError, pd.errors.EmptyDataError, pd.errors.ParserError, UnicodeDecodeError) as e:
+        raise ValueError(f"Error loading file '{df_name}': {e}")
+
+def fill_missing_values_shop_sales(df):
+    df['date_id'] = pd.to_numeric(df['date_id'], errors='coerce')
+    df['cnt'] = pd.to_numeric(df['cnt'], errors='coerce')
+    df[['item_id', 'store_id']] = df[['item_id', 'store_id']].astype(str)
+
+    max_date_id = df['date_id'].max()
+    for i in range(1, len(df)):
+        if pd.isna(df.at[i, 'date_id']):
```

Comment:
I'm wondering why you're assuming that all the data in the DataFrame will follow the pattern ID = [..., 7, 8, 9, ...]. What if the sequence of date_ids is like [..., 7, 8, 9, NaN, 120]?

> Why do you choose NaN to be 10 and not 119? It could also be 119, couldn't it?